lib/dialects/mariadb/query.js

```diff
@@ -0,0 +1,358 @@
+'use strict';
+
+const Utils = require('../../utils');
+const logger = require('../../utils/logger');
+const debug = logger.getLogger().debugContext('sql:mariadb');
+const AbstractQuery = require('../abstract/query');
+const uuidv4 = require('uuid/v4');
+const sequelizeErrors = require('../../errors');
+const _ = require('lodash');
+const DataTypes = require('../../data-types');
+
+const ER_DUP_ENTRY = 1062;
+const ER_ROW_IS_REFERENCED = 1451;
+const ER_NO_REFERENCED_ROW = 1452;
+
+class Query extends AbstractQuery {
+  constructor(connection, sequelize, options) {
+    super();
+    this.connection = connection;
+    this.instance = options.instance;
+    this.model = options.model;
+    this.sequelize = sequelize;
+    this.uuid = uuidv4();
+    this.options = Object.assign({
+      logging: console.log,
+      plain: false,
+      raw: false,
+      showWarnings: false
+    }, options || {});
+
+    this.checkLoggingOption();
+  }
+
+  static formatBindParameters(sql, values, dialect) {
+    const bindParam = [];
+    const replacementFunc = (match, key, val) => {
+      if (val[key] !== undefined) {
+        bindParam.push(val[key]);
+        return '?';
+      }
+      return undefined;
+    };
+    sql = AbstractQuery.formatBindParameters(sql, values, dialect,
+      replacementFunc)[0];
+    return [sql, bindParam.length > 0 ? bindParam : undefined];
+  }
+
+  run(sql, parameters) {
+    this.sql = sql;
+
+    //do we need benchmark for this query execution
+    const benchmark = this.sequelize.options.benchmark
+      || this.options.benchmark;
+    const showWarnings = this.sequelize.options.showWarnings
+      || this.options.showWarnings;
+
+    let queryBegin;
+    if (benchmark) {
+      queryBegin = Date.now();
+    } else {
+      this.sequelize.log(
+        `Executing (${this.connection.uuid || 'default'}): ${this.sql}`,
+        this.options);
+    }
+
+    debug(`executing(${this.connection.uuid || 'default'}) : ${this.sql}`);
+
+    if (parameters) {
+      debug('parameters(%j)', parameters);
+    }
+    return new Utils.Promise((resolve, reject) => {
+      this.connection.query(this.sql, parameters)
+        .then(results => {
+          debug(
+            `executed(${this.connection.uuid || 'default'}) : ${this.sql}`);
+
+          if (benchmark) {
+            this.sequelize.log(
+              `Executed (${this.connection.uuid || 'default'}): ${this.sql}`,
+              Date.now() - queryBegin, this.options);
+          }
+
+          // Log warnings if we've got them.
+          if (showWarnings && results && results.warningStatus > 0) {
+            resolve(this.logWarnings(results));
+            return;
+          }
+          resolve(results);
+        })
+        .catch(err => {
+          debug(
```

Comment:
Is it really useful to debug log and regular log?

Reply:
hmm, a better thing may be 
```
debug( `executed(${this.connection.uuid || 'default'}) with error : ${this.sql}\n${err}`);
```

Reply:
No keep this format